lib/browser/index.js

```diff
@@ -106,6 +106,15 @@ export default class Realm {
         let method = util.createMethod(objectTypes.REALM, 'objects');
         return method.apply(this, [type, ...args]);
     }
+
+    objectForPrimaryKey(type, ...args) {
+        if (typeof type == 'function') {
```

Comment:
We could just move the check for function into `typeForConstruction` to make this a one liner every time we need to do this.